js/src/Zone.js

```diff
@@ -2,57 +2,62 @@ import ExpectedMark from "./ExpectedMark";
+ * to a given phase of page load. (i.e. all elements in `ux-destination-verfied`)
  *
- * Only one set of zones can be tracked for the same view at one point in time
+ * Example config object:
+ *
+ * {
+ *  name: "ux-destination-verified",
+ *  marks: ["ux-image-online-logo", "ux-image-inline-logo"]
+ *  selectors: ["img.logo"],      // TODO: not yet implemented
+ *  startMark: "navigationStart",
+ *  onMeasure: measureName => {}
+ *  onMark: markName => {}
+ * }
  */
 export default class Zone {
-  // constructs individual zone object
+  /**
+   * Creates a Zone.
+   * @param {object} config
+   */
   constructor(config) {
-    // {
-    //   name: "ux-destination-verified",
-    //   marks: ["ux-image-online-logo", "ux-image-inline-logo"]
-    //   selectors: ["img.logo"],
-    //   startMark: "navigationStart",
-    //   onMeasure: measureName => {}
-    //   onMark: markName => {}
-    // }
-
-    // name to be used for UserTiming measures
+    // Name used for UserTiming measures
     this.measureName = config.name;
 
-    // handling deprecated "label" keys in backwards-compatible way
+    // Handle deprecated "label" keys
     if (config.label) {
-      console.warn(
-        "[ux-capture] Deprecation Warning: `label` keys on configuration object were renamed to `name` as of verision v2.0.0",
-        "Will be removed in v3.0.0"
-      );
+      if (console) {
+        console.warn(
+          "[ux-capture] Deprecation Warning: `label` keys on configuration object were renamed to `name` as of verision v2.0.0",
+          "Will be removed in v3.0.0"
+        );
+      }
 
       if (!config.name) {
         this.measureName = config.label;
       }
     }
 
-    // callback to execute when Zone is complete
+    // Callback to execute when Zone is complete (all marks triggered)
     this.onMeasure = config.onMeasure;
 
-    // callback for marks to call when they are complete
+    // Callback for marks to call when they are complete
     this.onMark = config.onMark;
 
-    this.startMarkName = config.startMarkName || "navigationStart";
-
-    // look up existing mark object or create a new one
+    // Mark used as starting point for subsequent UserTiming mesaures
+    // For Page View (initial page loads), this will be "navigationStart"
+    // For Interactive Views, this will be "interactionStart"
+    this.startMark = config.startMark;
 
-    // now just string names for UserTiming marks
-    // in the future, different types of events, e.g. ImageElement("logo"), PaintTimer("first-paint")
+    // Create a new `ExpectedMark` for each mark
+    // supplied in `config.marks`
     this.marks = config.marks.map(markName => {
       const mark = ExpectedMark.create(markName);
 
       mark.onComplete(mark => {
-        // if Zone's onMark callback is specified, call it with mark name
-        if (this.onMark) {
-          this.onMark(mark.name);
-        }
+        // Call Zone's `onMark` callback
+        this.onMark(mark.name);
```

Comment:
FYI - The default is NOOP, so `this.onMark` and `this.onMeasure` will always exist. So I removed the unnecessary checks.

Reply:
I was unsure if I should do that as Zone is somewhat isolated class so I thought that the fact of View always having a callback event if it's a NOOP is not Zone's concern.

Anyway, I have no problem going this route - if I had a strong feeling about it, I would've written some tests for it ;)

Reply:
When we write up the API spec for `Zone` (or eventually add type annotations), we can make the `onMark` + `onMeasure` params non-nullable.

alternatively, we could add the NOOP fallback in the `this.onMark/onMeasure` assignment in the constructor here as well, but that probably creeps toward overengineering. treating `Zone` as sort of a 'private' class that we have full control over should be fine